src/__test__/redux/actions/samples/updateSample.test.js

```diff
@@ -57,4 +73,48 @@ describe('updateSample action', () => {
 
     expect(saveSamples).toHaveBeenCalled();
   });
+
+  it('Dispatches events correctly for api v2', async () => {
+    config.currentApiVersion = api.V2;
+    fetchMock.mockResponseOnce(() => Promise.resolve(JSON.stringify({})));
+
+    const sampleDiff = {
+      name: 'updated name',
+    };
+
+    const store = mockStore(mockState);
+    await store.dispatch(updateSample(mockUuid, sampleDiff));
+
+    const actions = store.getActions();
+    expect(_.map(actions, 'type')).toEqual([SAMPLES_SAVING, SAMPLES_SAVED, SAMPLES_UPDATE]);
+    expect(_.map(actions, 'payload')).toMatchSnapshot();
+
+    expect(fetchMock).toHaveBeenCalledWith(
+      'http://localhost:3000/v2/experiments/null/samples/asd123',
+      {
+        body: JSON.stringify(sampleDiff),
+        headers: { 'Content-Type': 'application/json' },
+        method: 'PATCH',
+      },
+    );
+
+    expect(saveSamples).not.toHaveBeenCalled();
+  });
+
+  it('Error handling for api v2 works', async () => {
+    config.currentApiVersion = api.V2;
+
+    fetchMock.mockRejectOnce(() => Promise.reject(new Error('Api error')));
+
+    const sampleDiff = {
+      name: 'updated name',
+    };
+
+    const store = mockStore(mockState);
+    await store.dispatch(updateSample(mockUuid, sampleDiff));
+
+    const actions = store.getActions();
+    expect(_.map(actions, 'type')).toEqual([SAMPLES_SAVING, SAMPLES_ERROR]);
+    expect(_.map(actions, 'payload')).toMatchSnapshot();
```

Comment:
I like using these maps together with snapshot. I think it will make tests much easier to maintain and update if we do that everywhere.